### PR TITLE
Social logos | Add TypeScript support

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Currently, social-logos package doesn't have in-built TypeScript support, which makes it hard to allow auto-completion of icon names.

## Proposed changes:

* Rename React files to `.tsx` files
* Update React build step to use `tsc` for transpilation
* Sort auto generated icons 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Just run `pnpm run build` in `social-logos`
* Confirm that build is success and the files are generated correctly
* Confirm that React TSX files are transpiled correctly
* Using `pnpm link --global` expose the package globally and try to consume it in a fresh React project
* Confirm that the imports and auto-completions work as expected.

### CODE REVIEW

File: projects/js-packages/social-logos/src/svg/threads.svg [hunk 1]
```diff
@@ -1,3 +1,3 @@
...
...133.559 125.696 136.834 117.143 138.28 106.366C144.217 109.949 148.617 114.664 151.047 120.332C155.179 129.967 155.42 145.8 142.501 158.708C131.182 170.016 117.576 174.908 97.0135 175.059C74.2042 174.89 56.9538 167.575 45.7381 153.317C35.2355 139.966 29.8077 120.682 29.6052 96C29.8077 71.3178 35.2355 52.0336 45.7381 38.6827C56.9538 24.4249 74.2039 17.11 97.0132 16.9405C119.988 17.1113 137.539 24.4614 149.184 38.788C154.894 45.8136 159.199 54.6488 162.037 64.9503L178.184 60.6422C174.744 47.9622 169.331 37.0357 161.965 27.974C147.036 9.60668 125.202 0.195148 97.0695 0H96.9569C68.8816 0.19447 47.2921 9.6418 32.7883 28.0793C19.8819 44.4864 13.2244 67.3157 13.0007 95.9325L13 96L13.0007 96.0675C13.2244 124.684 19.8819 147.514 32.7883 163.921C47.2921 182.358 68.8816 191.806 96.9569 192H97.0695C122.03 191.827 139.624 185.292 154.118 170.811C173.081 151.866 172.51 128.119 166.26 113.541C161.776 103.087 153.227 94.5962 141.537 88.9883ZM98.4405 129.507C88.0005 130.095 77.1544 125.409 76.6196 115.372C76.2232 107.93 81.9158 99.626 99.0812 98.6368C101.047 98.5234 102.976 98.468 104.871 98.468C111.106 98.468 116.939 99.0737 122.242 100.233C120.264 124.935 108.662 128.946 98.4405 129.507Z"></path>
```
Comment (manzoorwanijk) [inline projects/js-packages/social-logos/src/svg/threads.svg:2]: I removed that `class="x19hqcy"` prop because TS won't like that.

File: projects/js-packages/social-logos/package.json [hunk 2]
```diff
@@ -34,10 +34,14 @@
...
 	},
 	"devDependencies": {
+		"@types/react": "18.3.3",
+		"@types/react-dom": "18.3.0",
+		"glob": "10.4.1",
 		"svg2ttf": "^6.0.3",
 		"svgicons2svgfont": "^12.0.0",
 		"svgo": "^3.3.2",
 		"svgstore-cli": "^2.0.1",
+		"typescript": "5.4.5",
```
Comment (anomiex) [inline projects/js-packages/social-logos/package.json]: If we don't need anything in the newer versions, I'd suggest matching the versions already in the monorepo, to avoid adding a bunch of extra entries into the lock file.
```suggestion
		"@types/react": "18.3.1",
		"@types/react-dom": "18.3.0",
		"glob": "10.3.15",
		"svg2ttf": "^6.0.3",
		"svgicons2svgfont": "^12.0.0",
		"svgo": "^3.3.2",
		"svgstore-cli": "^2.0.1",
		"typescript": "5.0.4",
```
(After changing that, you'll want to `git checkout trunk pnpm-lock.yaml` before doing `pnpm install` to get the cleanest resulting set of updates to the lock file)

If we do need newer versions of something, let me know and we can have Renovate update it for the whole monorepo easily enough.

File: projects/js-packages/social-logos/package.json [hunk 3]
```diff
@@ -8,8 +8,8 @@
 	},
 	"exports": {
 		".": {
-			"jetpack:src": "./src/react/index.jsx",
-			"default": "./build/react/index.jsx"
+			"jetpack:src": "./src/react/index.js",
```
Comment (anomiex) [inline projects/js-packages/social-logos/package.json]: Should this be
```suggestion
			"jetpack:src": "./src/react/index.ts",
```
?

File: projects/js-packages/social-logos/tools/svg-to-react-data.js
Comment (anomiex) [inline projects/js-packages/social-logos/tools/svg-to-react-data.js:1]: Hmm. I don't know how it managed to work before with no version of `glob` in `package.json`, but somehow it was picking up 7.1.6. Now with 10.3 or 10.4 it's returning the svgs in a different order, so we'll probably want to throw a `.sort()` in there somewhere before processing the files to make sure the data file is still consistent.

Comment (anomiex) [reply projects/js-packages/social-logos/tools/svg-to-react-data.js:1]: (Ah, it seems that when running `jetpack build` it winds up having `tools/cli/node_modules/` included in `NODE_PATH` so it finds the copy of `glob` from there if it's not in the project itself.)

Comment (tbradsha) [reply projects/js-packages/social-logos/tools/svg-to-react-data.js:1]: Oops, that's my oversight!

File: projects/js-packages/social-logos/package.json (hunk 2 above)
Comment (manzoorwanijk) [reply projects/js-packages/social-logos/package.json]: Done. Thanks

File: projects/js-packages/social-logos/package.json (hunk 3 above)
Comment (manzoorwanijk) [reply projects/js-packages/social-logos/package.json]: Yeah, thanks

File: projects/js-packages/social-logos/tools/svg-to-react-data.js
Comment (manzoorwanijk) [reply projects/js-packages/social-logos/tools/svg-to-react-data.js:1]: Added `.sort()` to `glob()` result.

**Comment by manzoorwanijk** [comment]
@tbradsha @anomiex I have addressed the feedback. Thank you both of you.

File: projects/js-packages/social-logos/tools/build-react [hunk 4]
```diff
@@ -11,8 +11,8 @@ cd "$root_dir"
 # Make dir if it doesn't exist.
 mkdir -p "$dest_react_dir"
 
-# Copy source files to build folder.
-cp "$src_react_dir/"* "$dest_react_dir"
+# Build files using TypeScript.
+pnpm tsc --noEmit false
```
Comment (anomiex) [inline projects/js-packages/social-logos/tools/build-react]: Why did you move the `--noEmit false` here instead of changing the tsconfig to extend `tsconfig.tsc.json` like the message told you to?

```suggestion
pnpm tsc
```

File: projects/js-packages/social-logos/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+	"extends": "jetpack-js-tools/tsconfig.base.json",
+	"compilerOptions": {
+		"declaration": true,
```
Comment (anomiex) [inline projects/js-packages/social-logos/tsconfig.json]:
```suggestion
	"extends": "jetpack-js-tools/tsconfig.tsc.json",
	"compilerOptions": {
```

**Review by anomiex — CHANGES_REQUESTED** [review]

File: projects/js-packages/social-logos/tools/build-react (hunk 4 above)
Comment (manzoorwanijk) [reply projects/js-packages/social-logos/tools/build-react]: May be I read that message wrongly. Thank you for the correction.

**Review by anomiex — APPROVED** [review]
LGTM now.

I do note that 3.0.0 accidentally changed the package from CommonJS to ESM, and this is now accidentally changing it back. Maybe that's good, but if we want to go ESM with the package after all we might consider setting `"type": "module"` in the package.json.

**Comment by tbradsha** [comment]
Note that this removes the `example.css` file from the build folder. I'll add it back in a future PR.
